pkg/agent/controller/agent.go

```diff
@@ -183,23 +224,31 @@ func (a *Controller) Start(stopCh <-chan struct{}) error {
 		return errors.Wrap(err, "error starting ServiceImport controller")
 	}
 
-	a.serviceExportSyncer.Reconcile(func() []runtime.Object {
-		return a.serviceImportLister(func(si *mcsv1a1.ServiceImport) runtime.Object {
-			return &mcsv1a1.ServiceExport{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      si.GetAnnotations()[lhconstants.OriginName],
-					Namespace: si.GetAnnotations()[lhconstants.OriginNamespace],
-				},
-			}
+	// check on startup if a local service export still exist for all remote service exports uploaded from this cluster.
+	// if not - enqueue deletion of the service export, to delete obsolete service export from the broker
+	a.serviceExportUploader.Reconcile(func() []runtime.Object {
+		return a.remoteServiceExportLister(func(se *mcsv1a1.ServiceExport) runtime.Object {
+			annotations := se.GetAnnotations()
+			// workaround a bug in resource_syncer.Reconcile() when in LocalToRemote mode:
+			// resource name on broker is not mapped back to resource name on origin cluster
+			se.Name = annotations[lhconstants.OriginName]
```

Comment:
1. is the original namespace not required as well? On Broker all SE's share a namespace and original value is present in labels/annotations only.
2. Any reason to prefer annotations over labels? They're currently redundant  and should be holding the same value. We should pick either labels or annotations exclusively in the code. In the `mcs` package used labels as the source of truth. The agent seems to use Annotations (e.g., lines around 250 set annotations).

Reply:
1. The original namespace is restored in resource_syncer recon func (https://github.com/submariner-io/admiral/blob/10c88bb61e69a881f4888b31061903fce867354c/pkg/syncer/resource_syncer.go#L366). This is just a workaround since they did not restore the original name only the namespace.
2. I also don't understand why they use both labels and annotations with redundant info. I preferred to do the least changes possible, but we can create an issue to consolidate labels and annotations. I prefer to make such a change after unit tests are passing again.